sdk/state/update.go

```diff
@@ -46,6 +46,10 @@ func (c *Channel) ProposePayment(amount Amount) (Payment, error) {
 	if amount.Amount <= 0 {
 		return Payment{}, errors.New("payment amount must be greater than 0")
 	}
+	if amount.Asset != c.latestCloseAgreement.Balance.Asset {
```

Comment:
need to validate the input asset is the one we're expecting since not always native now